File: chapter2/java/ToolbarFrame2.java
// ToolbarFrame2.java
// The Swing-ified button example
//
import java.awt.*;
import java.awt.event.*;
import javax.swing.*;

public class ToolbarFrame2 extends Frame {
	
	// This time, let's use JButtons!
	JButton cutButton, copyButton, pasteButton;
	JButton javaButton, macButton, motifButton, winButton;

	public ToolbarFrame2() {
		super("Toolbar Example (Swing)");
		setSize(450, 250);
		
		addWindowListener(new WindowAdapter(){
			public void windowClosing(WindowEvent e) {
				System.exit(0);
			}
		});

		ActionListener printListener = new ActionListener() {
			public void actionPerformed(ActionEvent ae) {
				System.out.println(ae.getActionCommand());
			}
		};

		// JPanel works similarly to Panel so we'll use it.
		JPanel toolbar = new JPanel();
		toolbar.setLayout(new FlowLayout(FlowLayout.LEFT));

		cutButton = new JButton("Cut");
		cutButton.addActionListener(printListener);
		toolbar.add(cutButton);

		copyButton = new JButton("Copy");
		copyButton.addActionListener(printListener);
		toolbar.add(copyButton);

		pasteButton = new JButton("Paste");
		pasteButton.addActionListener(printListener);
		toolbar.add(pasteButton);

		add(toolbar, BorderLayout.NORTH);

		// Add the L&F controls
		JPanel lnfPanel = new	JPanel();
		LnFListener lnfListener = new LnFListener(this);
		
		macButton = new JButton("Mac");
		macButton.addActionListener(lnfListener);
		lnfPanel.add(macButton);

		javaButton = new JButton("Metal");
		javaButton.addActionListener(lnfListener);
		lnfPanel.add(javaButton);
		
		motifButton = new JButton("Motif");
		motifButton.addActionListener(lnfListener);
		lnfPanel.add(motifButton);
		
		winButton = new JButton("Windows");
		winButton.addActionListener(lnfListener);
		lnfPanel.add(winButton);		

		add(lnfPanel, BorderLayout.SOUTH);
	}

	public static void main(String args[]) {
		ToolbarFrame2 tf2 = new ToolbarFrame2();
		tf2.setVisible(true);
	}

}

File: chapter2/jruby/ToolbarFrame2.rb
require 'java'

java_import java.awt.BorderLayout
java_import java.awt.FlowLayout
java_import java.awt.Frame

java_import javax.swing.JButton
java_import javax.swing.JPanel

# Events
java_import java.awt.event.ActionListener
java_import java.awt.event.WindowAdapter
java_import java.lang.System

# import java class
# java_import 'LnFListener'
require 'lnflistener'

class ToolbarFrame2 < Frame

	def initialize()
		super("Toolbar Example (Swing)")
		self.setSize 450, 250
	
		add_window_listener(WindowCloser.new)

		toolbar = JPanel.new
		toolbar.layout = FlowLayout.new(FlowLayout::LEFT)

		cut = JButton.new("Cut")
		cut.addActionListener(PrintListener.new)
		toolbar.add(cut)

		copy = JButton.new("Copy")
		copy.addActionListener(PrintListener.new)
		toolbar.add(copy)

		paste = JButton.new("Paste")
		paste.addActionListener(PrintListener.new)
		toolbar.add(paste)

		add(toolbar, BorderLayout::NORTH)

		# Add the L&F controls.
		lnfPanel = JPanel.new
		lnfListener = LnFListener.new(self)

		macButton = JButton.new('Mac')
		macButton.addActionListener(lnfListener)
		lnfPanel.add(macButton)

		javaButton = JButton.new('Metal')
		javaButton.addActionListener(lnfListener)
		lnfPanel.add(javaButton)
		
		motifButton = JButton.new('Motif')
		motifButton.addActionListener(lnfListener)
		lnfPanel.add(motifButton)
		
		winButton = JButton.new('Windows')
		winButton.addActionListener(lnfListener)
		lnfPanel.add(winButton)

		add(lnfPanel, BorderLayout::SOUTH)
		
	end

  class WindowCloser < WindowAdapter
    def window_closing
      System.exit()
    end
  end

  class PrintListener 
  	include ActionListener
  	def action_performed(ae)
  		System.out.println(ae.get_action_command)
  	end
  end

end

tf2 = ToolbarFrame2.new
tf2.visible = true